cip-observe-script-type/README.md

```diff
@@ -0,0 +1,201 @@
+
+### Native Script Extension
+
+The BNF notation for the abstract syntax of native scripts change as follows to reflect the new field.
+
+```BNF
+<native_script> ::=
+             <RequireSignature>  <vkeyhash>
+           | <RequireObserver>   <scripthash>
+           | <RequireTimeBefore> <slotno>
+           | <RequireTimeAfter>  <slotno>
+
+           | <RequireAllOf>      <native_script>*
+           | <RequireAnyOf>      <native_script>*
+           | <RequireMOf>        <num> <native_script>*
+```
+
+Native scripts are typically represented in JSON syntax. We propose the following JSON representation for the `RequireObserver` constructor:
+```JSON
+{
+  "type": "observer",
+  "keyHash": "OBSERVATION_SCRIPT_HASH"
+}
+```
+
+
+## Rationale: how does this CIP achieve its goals?
+
+Currently Plutus scripts (and native scripts) in a transaction will only execute when the transaction performs the associated ledger action (ie. a Plutus minting policy will only execute if the transaction mints or burns tokens with matching currency symbol). The only exception is the withdraw zero trick which relies on an obscure mechanic where zero amount withdrawals are not filtered by the ledger. Now using `required_observers` we can specify a list of scripts (supports both native and Plutus scripts) to be executed in the transaction independent of any ledger actions. The newly introduced `txInfoObservations` field in the script context provides a straightforward way for scripts to check that "a particular script validated this transaction".
+
+This change is not backwards-compatible and will need to go into a new Plutus language version.
+
+### Alternatives 
+
+- We could decide to accept the withdraw-zero staking script trick as an adequate solution, and just preserve the nonsensical withdraw zero case in future language versions. 
+- The staking script trick could be abstracted away from the developer by smart contract languages that compile to UPLC. 
+    - This can be dangerous since by distancing the developer from what is actually happening you open up the door for the developer to act on misguided assumptions. 
+
+## Path to Active
+
+### Acceptance Criteria
+- [] Fully implemented in Cardano.
+      
+### Implementation Plan
+- [] Passes all requirements of both Plutus and Ledger teams as agreed to improve Plutus script efficiency and usability.
```

Comment:
p.s. to https://github.com/cardano-foundation/CIPs/pull/749#discussion_r1585200445 ... changing the other tickbox to proper GitHub format. 🤓
```suggestion
- [ ] Passes all requirements of both Plutus and Ledger teams as agreed to improve Plutus script efficiency and usability.
```